Objects/object.c

```diff
@@ -887,6 +887,38 @@ PyObject_GetAttr(PyObject *v, PyObject *name)
     return NULL;
 }
 
+PyObject *
+_PyObject_GetAttrWithoutError(PyObject *v, PyObject *name)
+{
+    PyTypeObject *tp = Py_TYPE(v);
+    PyObject *ret = NULL;
+
+    if (!PyUnicode_Check(name)) {
+        PyErr_Format(PyExc_TypeError,
+                     "attribute name must be string, not '%.200s'",
+                     name->ob_type->tp_name);
+        return NULL;
+    }
+    if (tp->tp_getattro != NULL) {
+        if (tp->tp_getattro == PyObject_GenericGetAttr) {
+            ret = _PyObject_GenericGetAttrWithDict(v, name, NULL, 1);
+        }
+        else {
+            ret = (*tp->tp_getattro)(v, name);
+        }
+    }
+    else if (tp->tp_getattr != NULL) {
```

Comment:
It looks like there is some code duplication with ``PyObject_GetAttr``, does it make sense to factor it out?

Reply:
LGTM.

Reply:
@methane Yes this idea looks good.